src/main/java/de/tum/in/www1/artemis/web/rest/TextAssessmentEventResource.java

```diff
@@ -0,0 +1,108 @@
+package de.tum.in.www1.artemis.web.rest;
+
+import static de.tum.in.www1.artemis.web.rest.util.ResponseUtil.forbidden;
+
+import java.net.URISyntaxException;
+import java.util.List;
+import java.util.Optional;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+
+import de.tum.in.www1.artemis.domain.Course;
+import de.tum.in.www1.artemis.domain.User;
+import de.tum.in.www1.artemis.domain.analytics.TextAssessmentEvent;
+import de.tum.in.www1.artemis.repository.CourseRepository;
+import de.tum.in.www1.artemis.repository.TextAssessmentEventRepository;
+import de.tum.in.www1.artemis.repository.UserRepository;
+import de.tum.in.www1.artemis.service.AuthorizationCheckService;
+import de.tum.in.www1.artemis.web.rest.errors.EntityNotFoundException;
+
+/**
+ * REST controller for managing TextAssessmentEventResource.
+ */
+@RestController
+@RequestMapping("/analytics/text-assessment-events")
+public class TextAssessmentEventResource {
+
+    private final Logger log = LoggerFactory.getLogger(TextAssessmentEventResource.class);
+
+    private final TextAssessmentEventRepository textAssessmentEventRepository;
+
+    private final AuthorizationCheckService authCheckService;
+
+    private final UserRepository userRepository;
+
+    private final CourseRepository courseRepository;
+
+    @Value("${info.text-assessment-analytics-enabled}")
+    private Optional<Boolean> textAssessmentAnalyticsEnabled;
+
+    public TextAssessmentEventResource(TextAssessmentEventRepository textAssessmentEventRepository, AuthorizationCheckService authCheckService, UserRepository userRepository,
+            CourseRepository courseRepository) {
+        this.textAssessmentEventRepository = textAssessmentEventRepository;
+        this.authCheckService = authCheckService;
+        this.userRepository = userRepository;
+        this.courseRepository = courseRepository;
+    }
+
+    /**
+     * The text assessment analytics are enabled when the configuration info.text-assessment-analytics-enabled is set to true.
+     * A non existing entry or false mean that the text assessment analytics is not enabled
+     * @return whether the text assessment analytics are enabled or not
+     */
+    private boolean isTextAssessmentAnalyticsEnabled() {
+        return textAssessmentAnalyticsEnabled.isPresent() && Boolean.TRUE.equals(textAssessmentAnalyticsEnabled.get());
+    }
+
+    /**
+     * This function retrieves all of the events from the 'text_assessment_event' table
+     * @return returns a List of TextAssessmentEvent's
+     */
+    @GetMapping("/all")
```

Comment:
still the wrong REST URL, get works directly on the URL in plural, e.g.

@GetMapping("/courses") would allow you to download all courses

Reply:
Also I suggest that you filter after courses or exercises, otherwise too many events will be loaded which might lead to problems